enclave/src/main/java/org/hyperledger/besu/enclave/Enclave.java

```diff
@@ -110,18 +115,17 @@ public String deletePrivacyGroup(final DeletePrivacyGroupRequest content) {
   }
 
   private <T> T handleJsonResponse(
-      final int statusCode, final byte[] body, final Class<T> responseType) {
-    try {
-      if (statusCode == 200) {
-        return objectMapper.readValue(body, responseType);
-      } else {
-        final ErrorResponse errorResponse = objectMapper.readValue(body, ErrorResponse.class);
-        throw new EnclaveException(errorResponse.getError());
-      }
-    } catch (final JsonParseException | JsonMappingException e) {
-      throw new EnclaveException("Failed to deserialise received json", e);
-    } catch (final IOException e) {
-      throw new EnclaveException("Decoding Json stream failed", e);
+      final int statusCode, final byte[] body, final Class<T> responseType) throws IOException {
+
+    if (statusCode == 200) {
+      return objectMapper.readValue(body, responseType);
+    } else if (statusCode < 500) {
+      final ErrorResponse errorResponse = objectMapper.readValue(body, ErrorResponse.class);
+      throw new EnclaveException(errorResponse.getError());
+    } else {
+      final String utf8EncodedBody = new String(body, StandardCharsets.UTF_8);
```

Comment:
does this  imply the error response will not be a json formatted string? If so - maybe the function can be renamed to simply "handleResponse"

Reply:
Rather confusingly the error from orion will be json then we parse it, create a new class of `EnclaveException` with just the error message. Then read that error message and create a different response type and message when users request it via the api. I didn't want to make too many changes here. :smiley: